### PR TITLE
Feature/folder support

### PR DESCRIPTION
**Pull request by veerabio**
Based on the previous pull request #447  for folder support.

**What kind of change does this PR introduce?**

- [ ] bugfix
- [x] feature
- [ ] refactoring / style
- [ ] build / chore
- [ ] documentation

**Did you add tests for your changes?**

- [x] Yes, my code is well tested
- [ ] Not relevant

**If relevant, did you update the documentation?**

- [ ] Yes, I've updated the documentation
- [x] Not relevant

**Summary**

Supports dropping a folder.

**Does this PR introduce a breaking change?**
No

**Other information**
The package size has increased.

### CODE REVIEW

**Comment by veerabio** [comment]
Pull-request https://github.com/quarklemotion/html5-file-selector/pull/2 should fix the one failing test

**Comment by okonet** [comment]
I’d be interested to see a solution with the small bundle size

**Comment by veerabio** [comment]
@okonet please see my other pull-request

**Comment by okonet** [comment]
Hmm, I’m wondering if this can be optional somehow. Or dynamically loaded only if needed?

**Comment by quarklemotion** [comment]
@okonet yeah I think making the full mime DB lookup optional makes sense.  I have time to work on this later this week - I was thinking of adding a second optional options object parameter to `getDroppedOrSelectedFiles` where the mime DB could be passed in as an injected dependency.  Will post here when I have the new version published!

File: package.json [hunk 1]
```diff
@@ -75,6 +75,7 @@
   },
   "dependencies": {
     "attr-accept": "^1.0.3",
+    "html5-file-selector": "^1.0.1",
```
Comment (quarklemotion) [inline package.json]: please update to "^2.0.1" to reduce the bundle size impact
https://bundlephobia.com/result?p=html5-file-selector@2.0.1

**Comment by veerabio** [comment]
Below is the package size with the new version:
  dist/index.js
  Package has exceeded the size limit
  Package size: 31.68 KB
  Size limit:   5.5 KB

  dist/es/index.js
  Package has exceeded the size limit
  Package size: 8.81 KB
  Size limit:   3 KB

**Comment by quarklemotion** [comment]
@veerabio I made one more update to trim down the `html5-file-selector` size (removed babel-runtime dependency).  I just published v2.1.0 with this change which brings the npm module size down to 1.9kb minified!
https://bundlephobia.com/result?p=html5-file-selector@2.1.0

I'm hoping that will bring the total react-dropzone size to under the current limits?

**Comment by okonet** [comment]
See related work at #616 and related issue #609

**Pull request closed without merging**